lib/babe/verify_test.go

```diff
@@ -2,3 +2,1236 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 package babe
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func newTestHeader(t *testing.T, digest ...scale.VaryingDataTypeValue) *types.Header {
+	t.Helper()
+	header := types.NewEmptyHeader()
+	for _, d := range digest {
+		err := header.Digest.Add(d)
+		assert.NoError(t, err)
+	}
+
+	return header
+}
+
+func signAndAddSeal(t *testing.T, kp *sr25519.Keypair, header *types.Header, data []byte) error {
+	t.Helper()
+	sig, err := kp.Sign(data)
+	assert.NoError(t, err)
+
+	return header.Digest.Add(types.SealDigest{
```

Comment:
why do you `assert.noError` for some on line 35, but not in the actual return?  Seems odd that we make sure no errors are returned here, but the function signature still returns an error, and then `assert.NoError` on the calling side.

Reply:
Well my thinking was that DIgest.Add returns an error so I should make sure no errors before that function call, then assert that the add executed correctly on the caller side. Do you think there is a better way to handle that?

Reply:
Just add another assert.NoError for the error returned by `header.Digest.Add`?

Also you might want to change the kp.Sign error assertion from `assert.NoError` to `require.NoError` to stop the test execution, since I don't think consecutive assertions would be relevant here